cli/src/wallet_cli/mod.rs

```diff
@@ -775,13 +772,22 @@ pub async fn mint_nft_command(
         None
     };
 
-    let nft_options = MintNftParams::new()
+    let mut nft_options = MintNftParams::new()
         .with_address(address)
-        .with_immutable_metadata(immutable_metadata)
-        .with_metadata(metadata)
         .with_tag(tag)
         .with_sender(sender)
         .with_issuer(issuer);
+
+    if let Some(metadata) = metadata {
+        nft_options = nft_options.with_metadata(MetadataFeature::new([(metadata.clone(), metadata)])?);
```

Comment:
Same key and value?

Reply:
Well no, how should a user provide it? Do we want to support entries or only a single one?

Reply:
In the CLI I would support a single entry (at least for now), and I would even use a default key

Reply:
Added something